lib/plurimath/unitsml/parse.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require "parslet"
+module Plurimath
+  class Unitsml
+    class Parse < Parslet::Parser
+      rule(:units) { load_yaml("/constants/units.yaml", "units" ) }
+      rule(:prefixes) { load_yaml("/constants/prefixes.yaml", "prefixes" ) }
+      rule(:dimensions) { load_yaml("/constants/dimensions.yaml", "dimensions" ) }
+      rule(:sequence) { units | prefixes | match(/\s+|\d+|\w+/).as(:text) }
+      rule(:iteration) { sequence >> iteration | sequence }
+      rule(:expression) { iteration >> expression | iteration }
+
+      root :expression
+
+      def load_yaml(filepath, glob_var)
+        glob_var = "@@#{glob_var}"
+        if variable_defined?(glob_var)
+          get_saved_value(glob_var)
+        else
+          content = YAML.load_file(File.join(__dir__, filepath)).to_json
+          self.class.class_variable_set(
+            glob_var,
+            JSON.parse(content, symbolize_names: true)
```

Comment:
Why do we convert YAML into JSON?

Reply:
This is just for converting the keys from string to symbols, I’ll be moving this to a separate method something like `symbolize_keys`, and will be using a different approach there.

Reply:
we already have **[unitsml-ruby](https://github.com/unitsml/unitsml-ruby)  gem**  in progress, should we use that for this purpose?

Reply:
Yes please! The issue is https://github.com/unitsml/unitsml-ruby/issues/7